packages/components/src/snackbar/index.tsx

```diff
@@ -88,17 +94,24 @@ function UnforwardedSnackbar(
 
 	useSpokenMessage( spokenMessage, politeness );
 
-	// Only set up the timeout dismiss if we're not explicitly dismissing.
+	// The `onDismiss/onRemove` can have unstable references,
+	// trigger side-effect cleanup, and reset timers.
+	const callbackRefs = useRef( { onDismiss, onRemove } );
+	useLayoutEffect( () => {
```

Comment:
What's the reason for `useLayoutEffect` instead of `useEffect` ?  In theory `useEffect`s should be called in order, so we should be guaranteed that the `onDimiss` and `onRemove` refs are always up to date

Reply:
Nothing specific. This is just a common pattern; we even have [a `useLatestRef` hook](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/utils/hooks/use-latest-ref.ts), but I don't use here because returned value isn't recognized as Ref and linter complains.

Related: https://epicreact.dev/the-latest-ref-pattern-in-react/